src/java/org/apache/cassandra/locator/AbstractCloudMetadataServiceConnector.java

```diff
@@ -25,22 +25,39 @@
 import java.net.URL;
 import java.util.Map;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableMap;
 
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 abstract class AbstractCloudMetadataServiceConnector
 {
+    static final String METADATA_URL_PROPERTY = "metadata_url";
+
+    public static class DefaultCloudMetadataServiceConnector extends AbstractCloudMetadataServiceConnector
+    {
+        protected DefaultCloudMetadataServiceConnector(String metadataServiceUrl)
```

Comment:
as said before , what about make this metaserviceUrl with the full url path ,not only the url address.
Or make both zone path configurable which may be more flexible.

Reply:
It is metadata service address only because that is variable we configure in mocking tests. See `Ec2V2ConnectorMockingTest` in trunk.

So the idea is that we change just the service url but not the endpoints.